src/Symfony/Component/Translation/PseudoLocalizationTranslator.php

```diff
@@ -0,0 +1,273 @@
+                '1' => '①',
+                '2' => '②',
+                '3' => '③',
+                '4' => '④',
+                '5' => '⑤',
+                '6' => '⑥',
+                '7' => '⑦',
+                '8' => '⑧',
+                '9' => '⑨',
+                ':' => '∶',
+                ';' => '⁏',
+                '<' => '≤',
+                '=' => '≂',
+                '>' => '≥',
+                '?' => '¿',
+                '@' => '՞',
+                'A' => 'Å',
+                'B' => 'Ɓ',
+                'C' => 'Ç',
+                'D' => 'Ð',
+                'E' => 'É',
+                'F' => 'Ƒ',
+                'G' => 'Ĝ',
+                'H' => 'Ĥ',
+                'I' => 'Î',
+                'J' => 'Ĵ',
+                'K' => 'Ķ',
+                'L' => 'Ļ',
+                'M' => 'Ṁ',
+                'N' => 'Ñ',
+                'O' => 'Ö',
+                'P' => 'Þ',
+                'Q' => 'Ǫ',
+                'R' => 'Ŕ',
+                'S' => 'Š',
+                'T' => 'Ţ',
+                'U' => 'Û',
+                'V' => 'Ṽ',
+                'W' => 'Ŵ',
+                'X' => 'Ẋ',
+                'Y' => 'Ý',
+                'Z' => 'Ž',
+                '[' => '⁅',
+                '\\' => '∖',
+                ']' => '⁆',
+                '^' => '˄',
+                '_' => '‿',
+                '`' => '‵',
+                'a' => 'å',
+                'b' => 'ƀ',
+                'c' => 'ç',
+                'd' => 'ð',
+                'e' => 'é',
+                'f' => 'ƒ',
+                'g' => 'ĝ',
+                'h' => 'ĥ',
+                'i' => 'î',
+                'j' => 'ĵ',
+                'k' => 'ķ',
+                'l' => 'ļ',
+                'm' => 'ɱ',
+                'n' => 'ñ',
+                'o' => 'ö',
+                'p' => 'þ',
+                'q' => 'ǫ',
+                'r' => 'ŕ',
+                's' => 'š',
+                't' => 'ţ',
+                'u' => 'û',
+                'v' => 'ṽ',
+                'w' => 'ŵ',
+                'x' => 'ẋ',
+                'y' => 'ý',
+                'z' => 'ž',
+                '{' => '(',
+                '|' => '¦',
+                '}' => ')',
+                '~' => '˞', ]) : $text;
+        }
+
+        if (1.0 < $this->expansionFactor) {
+            $textVisibleLength = false === ($encoding = mb_detect_encoding($visibleText, null, true)) ? \strlen($visibleText) : mb_strlen($visibleText, $encoding);
+            $transMissingLength = (int) (ceil($textVisibleLength * $this->expansionFactor)) - $textVisibleLength;
+            if ($this->brackets) {
+                $transMissingLength -= 2;
+            }
+
+            if (0 < $transMissingLength) {
+                $transPrefixLength = (int) (floor($transMissingLength / 2));
+                $transPrefix = '';
+                for ($i = 0; $i < $transPrefixLength; ++$i) {
+                    $transPrefix .= 0 === $i % 2 ? '~' : ' ';
```

Comment:
Alternating between spaces and a single char is not the best way to expand length IMO. It is not representative of what would happen in languages with a longer text, as each space creates a soft-wrapping opportunity.

Reply:
I think you are right. What method would you apply? What about computing the shortest and the longest word of the translated string and then add `~` words of a random length between the two, until the desired expansion is met?

Reply:
I improved the expansion logic. It now take into account the original translation words lengths an their probability to produce a realistic expansion string.